trl/trainer/dpo_trainer.py

```diff
@@ -1170,6 +1170,11 @@ def concatenated_forward(self, model: nn.Module, batch: dict[str, Union[list, to
         per_token_logps[~loss_mask] = 0
         all_logps = per_token_logps.sum(-1)
 
+        if self.loss_type == "dpo_norm":
+            all_logps = (per_token_logps * loss_mask).sum(-1) / loss_mask.sum(-1)
+        else:
+            all_logps = per_token_logps.sum(-1)
+
```

Comment:
`per_token_logps` is already 0 where the loss mask is 0. See two lines above. So I guess here it's equivalent to

```suggestion
        if self.loss_type == "dpo_norm":
            all_logps /= loss_mask.sum(-1)
```

right?